src/components/result-catalog/__test__/ResultCatalog.test.tsx

```diff
@@ -0,0 +1,129 @@
+import { act, render, screen, waitFor } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import ResultCatalog from '../ResultCatalog';
+import { BrowserRouter, LinkProps, Route, Routes } from 'react-router-dom';
+import { AppContext } from '../../../stores/SearchContext';
+import userEvent from '@testing-library/user-event';
+
+jest.mock('../../../service/apiService');
+
+describe('ResultCatalog', () => {
+  const props = {
+    startPage: false,
+    handleStopSearch: () => {},
+  };
+  const apiData = {
+    results: [
+      {
+        id: 1,
+        name: 'Morty Smith',
+        gender: 'Male',
+        species: 'Human',
+        status: 'Alive',
+        image: 'https://rickandmortyapi.com/api/character/avatar/2.jpeg',
+      },
+    ],
+    total: 1,
+  };
+  jest.fn().mockResolvedValue(apiData);
+  const context = {
+    localStorageValue: '',
+    apiData: apiData,
+    setFetchData: () => {},
+    setLocalStorageValue: () => {},
+  };
+
+  test('renders the list', async () => {
+    await act(async () => {
+      render(
+        <BrowserRouter>
+          <AppContext.Provider value={context}>
+            <Routes>
+              <Route
+                path="/*"
+                element={<ResultCatalog {...props} />}
+              />
+            </Routes>
+          </AppContext.Provider>
+        </BrowserRouter>
+      );
+    });
+    expect(true).toBeTruthy();
```

Comment:
You can use in test screen.debug() for dev purposes and see what the component renders and make some assertions.
Or match snapshot

Reply:
will be fixed in the next PR